code/workspaces/infrastructure-api/src/auth/permissionMiddleware.js

```diff
@@ -32,42 +32,31 @@ export function projectPermissionWrapper(permissionSuffix) {
   };
 }
 
-export function permissionWrapper(requiredPermission) {
+export function permissionWrapper(requiredPermission, allowInstanceAdmin = true) {
+  const acceptedPermissions = [requiredPermission];
+  if (allowInstanceAdmin) acceptedPermissions.push(SYSTEM_INSTANCE_ADMIN);
```

Comment:
I don't think you need this, because the instanceAdmin role gives the SYSTEM_DATA_MANAGER permission, see https://github.com/NERC-CEH/datalab/blob/master/code/workspaces/auth-service/resources/permissions.yml#L5.

Reply:
This function is aimed at being wider than just the data manager role. It is being used for the instance admin role and there is another endpoint or two it is being used for with stacks if I remember correctly. This method allows it to be used easily in both scenarios. 

With regards to the data manager roll permission check, I was trying to be explicit in what permission we want for a particular endpoint. We want someone to be a data manager to use this endpoint. It just so happens that currently a instance admin also has that role, but I think this method would handle the potential of that changing in the future well: we would just need to remove data manager from under system admin in the permission yaml you linked to and the system would work as expected.

Reply:
Okay, so not actually needed for systemDataManagerPermissionWrapper (because it uses allowInstanceAdmin=false below), but potentially useful elsewhere.
Ideally, a user has roles, roles map to permissions from the permissions.yml, and the endpoints check permissions; but I can see you sometimes need an escape clause for the instanceAdmin role (which should be the only thing to give the SYSTEM_INSTANCE_ADMIN permission).